apps/advisor/pkg/app/checktyperegisterer/checktyperegisterer.go

```diff
@@ -121,7 +121,7 @@ func (r *Runner) Run(ctx context.Context) error {
 			},
 		}
 		for i := 0; i < r.retryAttempts; i++ {
-			err := r.createOrUpdate(ctx, logger, obj)
+			err := r.createOrUpdate(context.WithoutCancel(ctx), logger, obj)
```

Comment:
This is an attempt to fix the following error:

```
msg="Error creating check type, retrying" app=advisor.app runner=advisor.checktyperegisterer error="context canceled" attempt=4
```

I assume that the startup context gets eventually canceled and we may have not had time to register all the checks.

If the source of the error is something else, we should start seeing other kind of errors if we remove the `cancel` here from the context (e.g. if the API server is being shut down).